pkg/codegen/hcl2/model/type_union.go

```diff
@@ -120,6 +112,26 @@ func (t *UnionType) Traverse(traverser hcl.Traverser) (Traversable, hcl.Diagnost
 	}
 }
 
+// Equals returns true if this type has the same identity as the given type.
+func (t *UnionType) Equals(other Type) bool {
+	if t == other {
+		return true
+	}
+	otherUnion, ok := other.(*UnionType)
+	if !ok {
+		return false
+	}
+	if len(t.ElementTypes) != len(otherUnion.ElementTypes) {
+		return false
+	}
+	for i, t := range t.ElementTypes {
```

Comment:
Does this imply that the elements of the union are always sorted? Same question for sets.